### PR TITLE
Ns/versionize transparent

### PR DESCRIPTION
**Pull request by nsarlin-zama**
closes: https://github.com/zama-ai/tfhe-rs-internal/issues/747#issue-2550479942

### PR content/description
Add support for `#[repr(transparent)]` in the Versionize proc macro. This will skip the versioning of this type by versioning the inner type instead, similar to [serde](https://serde.rs/container-attrs.html#transparent).

In this pr I also removed some manual implementations of Versionize that were no longer needed.

### Check-list:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

File: tfhe/src/shortint/backward_compatibility/ciphertext/mod.rs
Comment (nsarlin-zama) [reply tfhe/src/shortint/backward_compatibility/ciphertext/mod.rs]: this shouldn't have been pushed, sorry !

**Review by tmontaigu — APPROVED** [review]

**Review by IceTDrinker — COMMENTED** [review]
Looks good, no need to re-approve for full tests, once this is green good to merge

**Pull request merged**
